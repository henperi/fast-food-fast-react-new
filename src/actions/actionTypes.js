export const OPEN_MODAL = 'OPEN_MODAL';
export const CLOSE_MODAL = 'CLOSE_MODAL';
export const SIGNUP_USER = 'SIGNUP_USER';
export const SET_AUTH_USER = 'SET_AUTH_USER';
export const REMOVE_AUTH_USER = 'REMOVE_AUTH_USER';
export const SET_ERRORS = 'SET_ERRORS';
export const GET_MENU = 'GET_MENU';
export const ADD_TO_CART = 'ADD_TO_CART';
export const REMOVE_FROM_CART = 'REMOVE_FROM_CART';
export const UPDATE_CART_ITEM_QUANTITY = 'UPDATE_CART_ITEM_QUANTITY';
export const TOGGLE_CART_SLIDER = 'TOGGLE_CART_SLIDER';
export const FETCH_CART_ITEMS = 'FETCH_CART_ITEMS';
export const SET_VISIBLE_CART = 'SET_VISIBLE_CART';
