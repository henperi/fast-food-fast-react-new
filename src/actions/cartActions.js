import {
  ADD_TO_CART, REMOVE_FROM_CART, TOGGLE_CART_SLIDER, SET_CURRENT_CART,
} from './actionTypes';

export const setCurrentCart = payload => ({
  type: SET_CURRENT_CART,
  payload,
});

export const addToCart = (id, name, price) => ({
  type: ADD_TO_CART,
  id,
  name,
  price,
});

export const removeFromCart = id => ({ type: REMOVE_FROM_CART, id });

export const updateCartQty = (id, updates) => ({
  type: 'UPDATE_CART_ITEM_QUANTITY',
  updates,
  id,
});

export const emptyCart = () => {
  localStorage.removeItem('foodItems');
  return { type: 'EMPTY_CART' };
};

export const toggleCartSlider = () => ({ type: TOGGLE_CART_SLIDER });
