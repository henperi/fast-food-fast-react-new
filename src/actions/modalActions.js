// import { OPEN_MODAL } from './actionTypes';

// export const openModal = ({
//   id, cat, title, price, description,
// } = {}) => ({
//   type: OPEN_MODAL,
//   modal: {
//     isOpen: true,
//     id,
//     cat,
//     title,
//     price,
//     description,
//   },
// });

// export const emptyCart = () => ({ type: 'EMPTY_CART' });
