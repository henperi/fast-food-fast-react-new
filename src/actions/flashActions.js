import { SHOW_FLASH, REMOVE_FLASH } from './actionTypes';

export const showFlash = (data = {}) => {
  const { title = '', message = '', flashType = 'success' } = data;
  return {
    type: SHOW_FLASH,
    payload: {
      title,
      message,
      flashType,
    },
  };
};

export const removeFlash = () => ({
  type: REMOVE_FLASH,
});
