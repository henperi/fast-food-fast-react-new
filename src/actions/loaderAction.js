import { TOGGLE_LOADER } from './actionTypes';

const toggleLoader = () => ({
  type: TOGGLE_LOADER,
});

export default toggleLoader;
