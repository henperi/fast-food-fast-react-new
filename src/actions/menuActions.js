import { SET_MENU, SET_ERRORS, SET_FEATURED_MENU } from './actionTypes';
import axiosInstance from '../utils/axiosInstance';

export const setMenu = (payload = {}) => ({
  type: SET_MENU,
  payload,
});

export const setFeaturedMenu = (payload = {}) => {
  const featuredMenu = payload.slice(0, 3);

  return {
    type: SET_FEATURED_MENU,
    payload: featuredMenu,
  };
};

export const getMenu = () => (dispatch) => {
  axiosInstance
    .get('/menu')
    .then((response) => {
      const { data } = response;
      dispatch({
        type: SET_MENU,
        payload: { menu: [...data.menu] },
      });
      dispatch(setFeaturedMenu(data.menu));
    })
    .catch((errors) => {
      const { response = {}, request } = errors;
      console.log(response);
      console.log('=======');
      console.log(request);

      if (response.data) {
        return dispatch({
          type: SET_ERRORS,
          payload: [...response.data.errors],
        });
      }

      return dispatch({
        type: SET_ERRORS,
        payload: [{ msg: 'An unknown error occurred, Please check your network and try again' }],
      });
    });
};

export const updateMenu = () => {};
