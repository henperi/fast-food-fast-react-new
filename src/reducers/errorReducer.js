import { SET_ERRORS } from '../actions/actionTypes';

const errorReducer = (state = [], action) => {
  switch (action.type) {
    case SET_ERRORS:
      return [...action.payload];

    default:
      return state;
  }
};

export default errorReducer;
