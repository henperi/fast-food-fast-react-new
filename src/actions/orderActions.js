import { SET_ERRORS } from './actionTypes';
import axiosInstance from '../utils/axiosInstance';
import { emptyCart } from './cartActions';

const placeOrder = (cartItems = {}, handleModal) => (dispatch) => {
  console.log(cartItems);
  const orderData = {
    foodItems: cartItems,
  };
  axiosInstance
    .post('/orders', orderData)
    .then((response) => {
      const { data } = response;
      console.log(data);
      handleModal();
      dispatch(emptyCart());
    })
    .catch((errors) => {
      const { response = {}, request } = errors;
      console.log(errors);
      console.log(response);
      console.log('=======');
      console.log(request);

      if (response.data) {
        return dispatch({
          type: SET_ERRORS,
          payload: [...response.data.errors],
        });
      }

      return dispatch({
        type: SET_ERRORS,
        payload: [{ msg: 'An unknown error occurred, Please check your network and try again' }],
      });
    });
};

export default placeOrder;
