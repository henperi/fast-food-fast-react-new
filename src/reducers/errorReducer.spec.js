/* eslint-disable no-undef */
import { SET_ERRORS } from '../actions/actionTypes';

// Reducer to be tested
import errorReducer from './errorReducer';

describe('errorReducer', () => {
  const initialState = [];
  it('should return the default state if no action is specified', () => {
    const action = { type: '' };
    expect(errorReducer(initialState, action)).toMatchSnapshot({});
    expect(errorReducer(initialState, action)).toEqual(initialState);
  });

  it('should update the state and set setErrors when type is SET_ERRORS and action is specified', () => {
    const action = {
      type: SET_ERRORS,
      payload: [{ name: 'invalid name', password: 'invalid password' }],
    };
    expect(errorReducer(initialState, action)).toMatchSnapshot({});
    expect(errorReducer(initialState, action)).toEqual([
      { name: 'invalid name', password: 'invalid password' },
    ]);
  });
});
