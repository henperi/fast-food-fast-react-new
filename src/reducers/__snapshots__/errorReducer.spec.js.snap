// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errorReducer should return the default state if no action is specified 1`] = `Object {}`;

exports[`errorReducer should update the state and set setErrors when type is SET_ERRORS and action is specified 1`] = `
Object {
  "0": Object {
    "name": "invalid name",
    "password": "invalid password",
  },
}
`;
